integration/update_field_test.go

```diff
@@ -83,3 +84,25 @@ func TestUpdateFieldSet(t *testing.T) {
 		})
 	}
 }
+
+func TestReplaceKeepOrder(t *testing.T) {
```

Comment:
It was tricky to implement compat test for that, so I've stayed with integration one. Please let me know if you see better option.

Reply:
```go
func TestReplaceKeepOrderCompat(t *testing.T) {
	setup.SkipForTigrisWithReason(t, "Schema violation")

	t.Parallel()

	s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{Providers: shareddata.Providers{shareddata.Scalars}})

	ctx := s.Ctx
	targetCollection := s.TargetCollections[0]
	compatCollection := s.CompatCollections[0]

	replace := bson.D{{"_id", int32(1)}, {"c", int32(1)}, {"b", int32(2)}, {"a", int32(3)}}

	_, err := targetCollection.InsertOne(ctx, bson.D{{"_id", 1}})
	require.NoError(t, err)
	_, err = compatCollection.InsertOne(ctx, bson.D{{"_id", 1}})
	require.NoError(t, err)

	_, err = targetCollection.ReplaceOne(ctx, bson.D{{"_id", 1}}, replace)
	require.NoError(t, err)
	_, err = compatCollection.ReplaceOne(ctx, bson.D{{"_id", 1}}, replace)
	require.NoError(t, err)

	targetResult := targetCollection.FindOne(ctx, bson.D{{"_id", 1}})
	require.NoError(t, targetResult.Err())
	compatResult := compatCollection.FindOne(ctx, bson.D{{"_id", 1}})
	require.NoError(t, compatResult.Err())

	var targetDoc, compatDoc bson.D
	require.NoError(t, targetResult.Decode(&targetDoc))
	require.NoError(t, compatResult.Decode(&compatDoc))

	assert.Equal(t, compatDoc, targetDoc)
}
```

Reply:
Does it actually make sens to initialize all providers just to insert the document with _id:1 and send replace query that will only match the _id:1?

Reply:
If you don't need providers for this test you can change the setup line to `s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{AddNonExistentCollection: true})`

Reply:
And it will initialize only with `Scalars`.

Reply:
Thanks!